Source/AssetRipper.SourceGenerated.Extensions/SerializedProgramExtensions.cs

```diff
@@ -7,18 +9,56 @@ public static class SerializedProgramExtensions
 		public static int GetTierCount(this ISerializedProgram program)
 		{
 			int tierCount = 1;
-			int tier = program.SubPrograms[0].ShaderHardwareTier;
-			for (int i = 1; i < program.SubPrograms.Count; i++)
+			if (program.Has_PlayerSubPrograms())
 			{
-				if (program.SubPrograms[i].ShaderHardwareTier <= tier)
+				// where is tier in PlayerSubProgram?
+			}
+			else
+			{
+				int tier = program.SubPrograms[0].ShaderHardwareTier;
+				for (int i = 1; i < program.SubPrograms.Count; i++)
 				{
-					break;
-				}
+					if (program.SubPrograms[i].ShaderHardwareTier <= tier)
+					{
+						break;
+					}
 
-				tierCount++;
+					tierCount++;
+				}
```

Comment:
Why was this changed? `ISerializedProgram.SubPrograms` and `ISerializedSubProgram.ShaderHardwareTier` are present on all Unity versions.

Reply:
`ISerializedProgram.SubPrograms` is always empty if `ISerializedProgram.PlayerSubProgram` is not null

Reply:
For now, let's just account for the situation where it's empty.